README.md

```diff
@@ -1,285 +1,382 @@
+
+The value provided by threshold is that it determines the lowest possible price that allows the trade to go through and close. However, if the prices drop below the limit, the transaction fails and everything is reverted. Following the example above, one can provide a value (in base units) such as `140 DAI`. This will allow the transaction to pass and then the trade will be deemed successful if the price of `MKR/DAI` does not drop below `1.4 MKR/DAI`.
+
+## OTC Interface Contract
+
+This contract provides an interface that exposes the following three methods:
+
+- `sellAllAmount`
+- `buyAllAmount`
+- `getPayAmount`
+
+All from the [Maker OTC Market](https://github.com/makerdao/maker-otc/blob/master/src/matching_market.sol).
+
+## Token Interface Contract
+
+This contract is an interface which contains `ERC-20 token` methods as well as methods from the [WETH token](https://github.com/dapphub/ds-weth/blob/master/src/weth9.sol) contract, such as `deposit` and `withdraw`.
+
+# [OasisDirectProxy Contract](https://github.com/makerdao/oasis-direct-proxy/blob/gg/readme/src/OasisDirectProxy.sol)
+
+**Summary:** This contract can be used by anyone who wishes to swap tokens directly on [eth2dai](https://www.notion.so/makerdao/%5B%3Chttps://eth2dai.com%3E%5D(%3Chttps://eth2dai.com/%3E)).
+
+## Key functionalities (as defined in the Smart Contract)
+
+### withdrawAndSend
+
+**Summary:** `withdrawAndSend` is used internally as the modifier states. This function is used in the following other methods:
+
+- `sellAllAmountBuyEth`
+- `buyAllAmountPayEth`
+- `buyAllAmountBuyEth`
+
+The purpose of this function is to withdraw the locked up `ETH` from the `WETH token` contract and then send it to the caller.
+
+**Arguments:**
+
+- `TokenInterface wethToken` - The address of the `WETH token` contract.
+- `uint wethAmt` - The amount which will be withdrawn and sent.
+
+**Modifiers**
+
+- `internal` - When you see this modifier, it means that the function and/or state variable can only be accessed internally (i.e. from within the current contract or contracts deriving from it).
+
+---
+
+### sellAllAmount
+
+**Summary:** `sellAllAmount` is used when triggering an ERC-20 for ERC-20 exchange, where the `payAmt` is sent to this contract from the caller. After that, the `otc` contract is guaranteed to have enough allowance in the `payToken` to do transfers on behalf of `this` contract. The next step is to proceed and sell the `payToken`. Lastly, the transfer occurs for the `buyAmt` to the caller.
```

Comment:
**Summary:** `sellAllAmount` is used when triggering an ERC-20 for ERC-20 exchange, where the `payAmt` is sent to the `this` (OasisDirectProxy) contract from the caller. After that, the `otc` contract is guaranteed to have enough allowance in the `payToken` to do transfers on behalf of `this` (OasisDirectProxy) contract. The next step is to proceed and sell the `payToken`. Lastly, the transfer occurs for the `buyAmt` to the caller.

Reply:
I believe it is worth defining what contract is actually talked about here. You guys can decide if it is worth using brackets or not.